src/manifest.loader.ts

```diff
@@ -4,13 +4,56 @@ import { logger } from './log';
 import { Manifest, ManifestFile } from './manifest';
 
 const BackupExtension = '.1';
+export const ManifestFileName = 'manifest.json';
 
 function isManifestPath(fileName: string): boolean {
   if (fileName.endsWith('.json' + BackupExtension)) return true;
   if (fileName.endsWith('.json')) return true;
   return false;
 }
 
+/**
+ * Compare the 'Files' data of two manifest.json files.
+ * manifestA can contain more files than manifestB.
+ *
+ * @param manifestA The manifest in the source directory.
+ * @param manifestB The manifest in the destination directory.
+ * @returns true if manifestA contains all the files (with identical data) of manifestB.
+ */
+export function isManifestsDifferent(manifestA: ManifestLoader, manifestB: ManifestLoader): boolean {
+  for (const value of manifestB.files.values()) {
+    const comparedValue = manifestA.files.get(value.path);
+
+    if (comparedValue == null) return true;
+    if (comparedValue.path !== value.path) return true;
+    if (comparedValue.size !== value.size) return true;
+    if (comparedValue.hash !== value.hash) return true;
```

Comment:
what about if

`manifestA.hash = null` and `manifestB.hash = 'sha265-...'`  this would fail?

Reply:
It will return true as null <> 'sha265-...'. I guess this is not what we want in case of the file has not been uploaded/the hash has not been updated during a previous sync due to a network issue. Could you please confirm?

Reply:
I think we need a if the hash exists remotely but doesnt exist locally it may not actually be a problem. maybe we should fail though that is a very weird state to be in.